src/main/java/edit_pending_list_use_case/EditPendingListInteractor.java

```diff
@@ -0,0 +1,50 @@
+package edit_pending_list_use_case;
+
+import Entities.Group;
+import Entities.User;
+
+public class EditPendingListInteractor implements EditPendingListInputBoundary {
+
+    final PendingListDsGateway dsGateway;
+    final PendingListPresenter presenter;
+
+    public EditPendingListInteractor(PendingListDsGateway dsGateway, PendingListPresenter presenter) {
+        this.dsGateway = dsGateway;
+        this.presenter = presenter;
+    }
+
+    public EditPendingListResponseModel addOrRemoveUser(EditPendingListRequestModel requestModel) {
+        Integer userID = requestModel.getUserID();
+        Integer groupID = requestModel.getGroupID();
+        boolean pendingStatus = requestModel.getPendingStatus();
+
+        if(!dsGateway.userExistsByID(userID)) {return presenter.prepareFailView("User does not exist.");}
+
+        // TODO: this shouldn't be possible, maybe make it an exception instead of presenting it to the user?
```

Comment:
You're so right about this actually.. in my use case I also checked for these weird border cases even though it they shouldn't be possible. Would it be better to make them exceptions? Having these explicit checks makes my interactor code a lot longer.

Reply:
Yes, I think having them as exceptions is a good idea.